### PR TITLE
chore(region-info): Add AppConfig Lambda Layer ARNs

### PR DESCRIPTION
**Pull request by zeyad001**
This PR adds the AWS AppConfig Lambda Layer ARNs (found [here in the AWS Docs](https://docs.aws.amazon.com/appconfig/latest/userguide/appconfig-integration-lambda-extensions-versions.html)) to the region information.

[Previous pull request](https://github.com/aws/aws-cdk/pull/26873)

----

*By submitting this pull request, I confirm that my contribution is made under the terms of the Apache-2.0 license*

### CODE REVIEW

**Review by lpizzinidev — CHANGES_REQUESTED** [review]
Thanks 👍
Looks good overall to me.
Just a couple of minor changes on documentation and a question on older versions.
Can you also please reformat the [PR description](https://github.com/aws/aws-cdk/pull/27176#issue-1900786699) to be more readable?

File: packages/aws-cdk-lib/region-info/build-tools/fact-tables.ts [hunk 1]
```diff
@@ -215,6 +215,68 @@ export const APPMESH_ECR_ACCOUNTS: { [region: string]: string } = {
   'us-west-2': '840364872350',
 };
 
+// https://docs.aws.amazon.com/appconfig/latest/userguide/appconfig-integration-lambda-extensions-versions.html
+export const APPCONFIG_LAMBDA_LAYER_ARNS: { [key: string]: any } = {
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/region-info/build-tools/fact-tables.ts:219]: Shouldn't [older versions](https://docs.aws.amazon.com/appconfig/latest/userguide/appconfig-integration-lambda-extensions-versions.html#appconfig-integration-lambda-extensions-enabling-older-versions) be added as well?

File: packages/aws-cdk-lib/region-info/lib/region-info.ts [hunk 2]
```diff
@@ -153,6 +153,15 @@ export class RegionInfo {
     return Fact.find(this.name, FactName.cloudwatchLambdaInsightsVersion(insightsVersion, architecture));
   }
 
+  /**
+   * The ARN of the AppConfig Lambda Layer, for the given version.
+   * @param layerVersion the version (e.g. 2.0.181)
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/region-info/lib/region-info.ts:158]: `The layer version` is clearer

File: packages/aws-cdk-lib/region-info/lib/region-info.ts [hunk 3]
```diff
@@ -153,6 +153,15 @@ export class RegionInfo {
...
+   * @param architecture The Lambda Function architecture (e.g. 'x86_64' or 'arm64')
```
Comment (lpizzinidev) [inline packages/aws-cdk-lib/region-info/lib/region-info.ts:159]: `, defaults to x86_64.` should be specified here as well.

File: packages/aws-cdk-lib/region-info/build-tools/fact-tables.ts (hunk 1 above)
Comment (zeyad001) [reply packages/aws-cdk-lib/region-info/build-tools/fact-tables.ts:219]: Added

**Comment by zeyad001** [comment]
Addressed requested changes and reformatted the pr body.

**Review by lpizzinidev — APPROVED** [review]
Looks good to me 👍
Thanks for the quick changes

**Review by mrgrain — APPROVED** [review]

**Pull request merged**